libraries/chain/controller.cpp

```diff
@@ -307,6 +309,13 @@ struct assembled_block {
          v);
    }
 
+   digest_type action_mroot() const {
+      return std::visit(
+         overloaded{[](const assembled_block_legacy& ab) -> digest_type { assert(false); return digest_type{}; },
+                    [](const assembled_block_if& ab)     -> digest_type { return ab.action_mroot; }},
+         v);
```

Comment:
If this is needed, then it should return `const digest_type&`

Reply:
I thought about `const digest_type&`. But for Legacy, it returns a reference to a temporary, which is ugly.